File: backend/app/schemas/group_member.py
from uuid import UUID

from pydantic import BaseModel, ConfigDict, EmailStr


class AddMemberRequest(BaseModel):
    email: EmailStr


class GroupMemberResponse(BaseModel):
    id: UUID
    user_id: UUID
    group_id: UUID

    full_name: str
    email: EmailStr

    role: str

    model_config = ConfigDict(from_attributes=True)

File: frontend/lib/features/groups/models/group_member.dart
class GroupMember {
  final String id;
  final String userId;
  final String fullName;
  final String email;
  final String role;

  GroupMember({
    required this.id,
    required this.userId,
    required this.fullName,
    required this.email,
    required this.role,
  });

  factory GroupMember.fromJson(Map<String, dynamic> json) {
    return GroupMember(
      id: json["id"],
      userId: json["user_id"],
      fullName: json["full_name"],
      email: json["email"],
      role: json["role"],
    );
  }
}